pkg/datasource/data.go

```diff
@@ -426,26 +428,67 @@ func (ds *dataSource) AddStaticFields(size uint32, fields []StaticField) (FieldA
 	}}, nil
 }
 
-func (ds *dataSource) applyFieldConfig(newFields ...*field) {
-	if ds.config == nil {
+func applyFieldTemplate(f *field, annotations map[string]string) {
+	templateAnn := annotations[TemplateAnnotation]
+	if templateAnn == "" {
 		return
 	}
 
-	fields := ds.config.Sub("fields")
-	if fields == nil {
+	template := annotationsTemplates[templateAnn]
+	if template == nil {
+		log.Warnf("template %q not found for field %q", templateAnn, f.Name)
 		return
 	}
 
+	log.Debugf("applying template %q to field %q", templateAnn, f.Name)
+	for k, v := range template {
+		f.Annotations[k] = v
+	}
+}
+
+func (ds *dataSource) applyFieldConfig(newFields ...*field) {
+	var fields *viper.Viper
+
+	if ds.config != nil {
+		fields = ds.config.Sub("fields")
+	}
+
+	// first pass applying configuration from metadata
 	for _, field := range newFields {
-		sub := fields.Sub(field.Name)
-		if sub == nil {
+		// apply template if setted
+		applyFieldTemplate(field, field.Annotations)
```

Comment:
This is when, for instance, an operator adding a new field, adds also a template annotation for it?

Reply:
Exactly. I added a comment that hopefully makes it more clear.